### DISCUSSION TITLE
How to ungroup dependencies (i.e. do not group same dependency into one PR)?

### DISCUSSION THREAD

**Discussion by mikaello** [renovatebot/renovate#20514] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Wanted end result.

I want to add some config to make all dependency updates in specific folders be created as separate pull requests, even if the same dependency exists with the same update in other parts of the repository.

Example repository:

```
.
├── production
│   └── file-with-dependencies.yml
└── staging
    └── file-with-dependencies.yml
```

If I got a dependency `X` in both `production/` and `staging/` parts of the repository, Renovate will by default group these into a single pull request, but I want these two updates for the exact same dependency to be two separate pull requests.

The use-case for this is that we use Flux to provision our Kubernetes environment, and we want to update the staging environment before we update production, hence we need to merge the updates to `staging/` first.

An acceptable solution would also be to disable grouping of dependencies altogether, not just for a specific folder.

### What you tried so far.

Adding config to group everything in `production` folder:

```json
  "packageRules": [
    {
      "groupName": "Production updates",
      "groupSlug": "prod",
      "matchPaths": [
        "production/**"
      ]
    }
  ]
```

This will separate updates for the same dependency in `staging/` and `production/`, but it will unfortunately group all dependencies in `production/` into the same pull request. I can prevent this by adding more specific rules, e.g. `matchPackagePatterns`, but this is not durable when the number of dependencies increases.

**Comment by viceice** [comment, marked as answer]
use additional branch prefix instead of grouping. 😉

**Reply by mikaello** [reply]
Thanks, that is perfect!

When changing my grouping config to 

```json
"additionalBranchPrefix": "{{parentDir}}-"
```

Renovate does not group dependencies that reside in different folders.
